src/main/java/ua/goit/gojava32/kickstarter/dao/AbstractDAO.java

```diff
@@ -0,0 +1,59 @@
+package ua.goit.gojava32.kickstarter.dao;
+
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public abstract class AbstractDAO<T> implements GenericCRUDDAO<T> {
+
+  Class<T> type;
+
+  AbstractDAO(Class<T> type){
+    this.type = type;
+  }
+
+  @Autowired
+  SessionFactory sessionFactory;
+
+  Session getSession(){
+    return sessionFactory.getCurrentSession();
+  }
+
+  @Override
+  public T add(T val) {
+    Logger logger = Logger.getLogger(this.getClass());
+    logger.info("ADD:::::::::" + val);
```

Comment:
Convention is:
1. to use 'ing' in statements happening before stuff
2. 'ed' in statements happening after stuff
3. debug level for such ordinary things
4. also google for logger {}-placeholders 
so the correct way of writing this is:

````
logger.debug("Adding {}", val);
session.save(val);
````